File: 3760-maximum-substrings-with-distinct-start/3760-maximum-substrings-with-distinct-start.cpp
class Solution {
public:
    int maxDistinct(string s) {
        vector<int> v(26,0);
        int n = (int)s.size();
        int c = 0;
        for(int i=0;i<n;++i){
            int j = s[i]-'a';
            if(!v[j]){
                v[j]=1;
                ++c;
            }
        }
        return c;
    }
};

File: 3760-maximum-substrings-with-distinct-start/3760-maximum-substrings-with-distinct-start.java
class Solution {
    public int maxDistinct(String s) {
        boolean[] seen=new boolean[26];
        int c=0;
        for(int i=0;i<s.length();i++){
            int x=s.charAt(i)-'a';
            if(!seen[x]){seen[x]=true;c++;}
        }
        return c;
    }
}